translib/transformer/test/transformer_test.go

```diff
@@ -0,0 +1,299 @@
+
+	"testing"
+)
+
+var dbConfig struct {
+	Instances map[string]map[string]interface{} `json:"INSTANCES"`
+	Databases map[string]map[string]interface{} `json:"DATABASES"`
+}
+var rclient *redis.Client
+var filehandle *os.File
+var ygSchema *ytypes.Schema
+var rclientDBNum map[db.DBNum]*redis.Client
+
+func getDBOptions(dbNo db.DBNum, isWriteDisabled bool) db.Options {
+        var opt db.Options
+
+        switch dbNo {
+        case db.ApplDB, db.CountersDB, db.AsicDB, db.FlexCounterDB:
+                opt = getDBOptionsWithSeparator(dbNo, "", ":", ":", isWriteDisabled)
+                break
+        case db.ConfigDB, db.StateDB, db.ErrorDB:
+                opt = getDBOptionsWithSeparator(dbNo, "", "|", "|", isWriteDisabled)
+                break
+        }
+
+        return opt
+}
+
+func getDBOptionsWithSeparator(dbNo db.DBNum, initIndicator string, tableSeparator string, keySeparator string, isWriteDisabled bool) db.Options {
+        return (db.Options{
+                DBNo:               dbNo,
+                InitIndicator:      initIndicator,
+                TableNameSeparator: tableSeparator,
+                KeySeparator:       keySeparator,
+                IsWriteDisabled:    isWriteDisabled,
+        })
+}
+
+func getAllDbs(isGetCase bool) ([db.MaxDB]*db.DB, error) {
+	var dbs [db.MaxDB]*db.DB
+	var err error
+	var isWriteDisabled bool
+
+	if isGetCase {
+		isWriteDisabled = true
+	} else {
+		isWriteDisabled = false
+	}
+
+	//Create Counter DB connection
+	dbs[db.CountersDB], err = db.NewDB(getDBOptions(db.CountersDB, isWriteDisabled))
+
+	if err != nil {
+		closeAllDbs(dbs[:])
+		return dbs, err
+	}
+        
+        //Create Config DB connection
+	dbs[db.ConfigDB], err = db.NewDB(getDBOptions(db.ConfigDB, isWriteDisabled))
+
+	if err != nil {
+		closeAllDbs(dbs[:])
+		return dbs, err
+	}
+
+	return dbs, err
+}
+
+// Closes the dbs, and nils out the arr.
+func closeAllDbs(dbs []*db.DB) {
+	for dbsi, d := range dbs {
+		if d != nil {
+			d.DeleteDB()
+			dbs[dbsi] = nil
+		}
+	}
+}
+
+func TestMain(t *testing.M) {
+	fmt.Println("----- Setting up transformer tests -----")
+	if err := setup(); err != nil {
+		fmt.Fprintf(os.Stderr, "Error setting up transformer testing state: %v.\n", err)
+		os.Exit(1)
+	}
+	t.Run()
+	teardown()
+	os.Exit(0)
+}
+
+
+func initDbConfig() error {
+	dbConfigFile := "/run/redis/sonic-db/database_config.json"
+	if path, ok := os.LookupEnv("DB_CONFIG_PATH"); ok {
+		dbConfigFile = path
+	}
+
+	fmt.Println("dbConfigFile =", dbConfigFile)
+	dbConfigJson, err := ioutil.ReadFile(dbConfigFile)
+	if err == nil {
+		err = json.Unmarshal(dbConfigJson, &dbConfig)
+	}
+
+	return err
+}
+
+func clearDb() {
+
+	tblList := []string{
+		"TEST_SENSOR_GROUP",
+		"TEST_SENSOR_A_TABLE",
+		"TEST_SENSOR_B_TABLE",
+		"TEST_SET_TABLE",
+		"TEST_SENSOR_MODE_TABLE",
+	}
+
+	for _, tbl := range tblList {
+		_, err := exec.Command("/bin/sh", "-c",
+			"sonic-db-cli CONFIG_DB del `sonic-db-cli CONFIG_DB keys '"+
```

Comment:
1) `sonic-db-cli` will not be available outside sonic switch. Please avoid such switch dependencies so that we can these tests for PR sanity verification. Can you use the redis connections from `rclientDBNum` map?

2) As per the sonic yang definition TEST_SENSOR_MODE_TABLE is in counters_db. But this cleanup step uses config_db

Reply:
Addressed as suggested

Reply:
I still see sonic-db-cli usage and TEST_SENSOR_MODE_TABLE operated on config_db

Reply:
Done. Added the corrected file missing in previous commit.